crates/biome_js_analyze/src/assists/source/sort_jsx_props.rs

```diff
@@ -0,0 +1,178 @@
+    /// This rule checks if the JSX props are sorted in a consistent way.
+    /// A spread prop resets the sorting order.
+    ///
+    /// The rule can be configured to sort props alphabetically, ignore case, and more.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// <Hello lastName="Smith" firstName="John" />;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// <Hello firstName="John" lastName="Smith" />;
+    /// <Hello tel={5555555} {...this.props} firstName="John" lastName="Smith" />;
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// ### `callbacksLast`
+    ///
+    /// When `true`, callback props are sorted last.
+    ///
+    /// #### Example
+    ///
+    /// ```js
+    /// <Hello tel={5555555} onClick={this._handleClick} />;
+    /// ```
+    ///
+    /// ### `shordhand`
+    ///
+    /// When `first`, shorthand props are sorted first.
+    /// When `last`, shorthand props are sorted last, unless `callbacksLast` is `true`,
+    /// in which case they are sorted before callbacks.
+    /// Default is `ignore`.
+    ///
+    /// #### Example
+    ///
+    /// ```js
+    /// // shorthand first
+    /// <Hello active validate name="John" tel={5555555} />;
+    /// // shorthand last
+    /// <Hello name="John" tel={5555555} active validate />;
+    /// ```
+    ///
+    /// ### `multiline`
+    ///
+    /// When `first`, multiline props are sorted first, unless `shorthand` is `first`,
+    /// in which case they are sorted after shorthand props.
+    /// When `last`, multiline props are sorted last, unless `shorthand` is `last` or `callbacksLast` is `true`,
+    /// in which case they are sorted before shorthand props or callbacks.
+    /// Default is `ignore`.
+    ///
+    /// #### Example
+    ///
+    /// ```js
+    /// // multiline first
+    /// <Hello
+    ///   classes={{
+    ///     greetings: classes.greetings
+    ///   }}
+    ///   name="John"
+    ///   tel={5555555}
+    ///   active
+    ///   validate
+    /// />;
+    /// // multiline last
+    /// <Hello
+    ///   name="John"
+    ///   tel={5555555}
+    ///   active
+    ///   validate
+    ///   classes={{
+    ///     greetings: classes.greetings
+    ///   }}
+    /// />;
+    /// ```
+    ///
+    pub SortJsxProps {
+        version: "next",
+        name: "sortJsxProps",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintReact("jsx-sort-props")],
+        source_kind: RuleSourceKind::SameLogic,
+    }
+}
+
+impl Rule for SortJsxProps {
+    type Query = Ast<JsxAttributeList>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(_ctx: &RuleContext<Self>) -> Self::Signals {
+        Some(())
+    }
+
+    fn action(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<JsRuleAction> {
+        let props = ctx.query().clone();
+        let mut non_spread_props: Option<Vec<_>> = None;
```

Comment:
`Option<Vec<>>` is not very easy to use. You can just use `Vec<>`, and if the vector is empty, it means it's like `None`